### PR TITLE
Lab 3 Preparation Changes 

### PR DESCRIPTION
**Pull request by vmaravind96**
- Added Github actions for Lab 3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+project/
 .ivy2
 .sbt
+.DS_Store
+.metals/
+.vscode/
+*/.DS_Store
+.ammonite
+.bloop
+metals.sbt
```
Comment (Abhishek-P) [inline .gitignore]: There are some excellent conventions for organizing the `.gitignore`

Folders followed by file extensions etc. 

Good to check it out

File: .github/workflows/sbt-compile.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          distribution: temurin
+          java-version: 11
+          cache: sbt
+
+      - name: Run sbt compile
+        timeout-minutes: 2
+        working-directory: ./student_code
+        run: sbt compile
+        
+
```
Comment (Abhishek-P) [inline .github/workflows/sbt-compile.yml]: extra new line

**Review by Abhishek-P — APPROVED** [review]

**Comment by bechang** [comment]
@vmaravind96 @chfont  We'll want to get these actions into the deployment from `pppl-labdev`. As currently set up, these repos are really generated from `pppl-labdev`, so we could lose these in upcoming semesters if we don't address this tech debt.

File: .gitignore (hunk 1 above)
Comment (vmaravind96) [reply .gitignore]: Okay noted. Will check it out.

File: .github/workflows/sbt-compile.yml (hunk 2 above)
Comment (vmaravind96) [reply .github/workflows/sbt-compile.yml]: Removed

**Review by bechang — APPROVED** [review]
This looks good to me. We'll still need to add the final lab3.pdf before we merge this into main. But we can do that separately.

**Comment by Abhishek-P** [comment]
@vmaravind96 The lab3 pdf has aesthetic formatting problem in page 5 -> 7 we basically have two pages of figs interupting the text. Would be good to fix long-term, please feel free to raise a tracking issue for the same.

**Pull request merged**